src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -330,6 +320,25 @@ private Map<Integer, Float> doANNSearch(final LeafReaderContext context, final B
             .collect(Collectors.toMap(KNNQueryResult::getId, result -> knnEngine.score(result.getScore(), spaceType)));
     }
 
+    @VisibleForTesting
+    List<String> getEngineFiles(SegmentReader reader, String extension) throws IOException {
```

Comment:
1. Should make this method to return `Path indexPath` of the engine file?
2. Is there a way to move this method outside of KNNWeight so that we can avoid `@VisibleForTesting`?